### DISCUSSION TITLE
Current status of OpenMM and LAMMPS integration

### DISCUSSION THREAD

**Discussion by svandenhaute** [ACEsuit/mace#69] [Q&A] [answered]
Given the recent OpenMM 8.0 release, what is the current status of MACE's compatibility with it? Both in terms of the systems it supports (periodic/nonperiodic) as well as the compute platforms (CPU/CUDA) and the actual speed gains with respect to the native ASE calculator and the faster `matscipy` neighbor list implementation (also as a function of compute platform).
There are a few different branches with `openmm` in their name but it's a little unclear which, if any, are actually supposed to be used at the moment.
What about the LAMMPS interface?

**Comment by ilyes319** [comment, marked as answer]
Hey! 
Thank you for your interest.
The opemMM interface is currently in beta on a different fork https://github.com/davkovacs/mace/tree/openmm3 and a pull request is open to merge all these functionalities. @davkovacs will know better, but from what I understand, it can currently do non-periodic ML and MM-ML simulations. More detailed tutorials will be available in the coming weeks. For periodic systems, we are yet to solve all the compatibility issues with openMM. You can do CUDA MD; the speed is 15-20 ns/day.
As for LAMMPs, we also have a CPU interface in beta that we are testing internally and developing the GPU one.
All these functionalities are programmed to be released in the coming months on the main branch as part of the mace 0.1 version, but early access will be on the dev branch.

**Reply by davkovacs** [reply]
Here is a working example for openmm, you don’t actually need that branch any more. 

https://github.com/sef43/openmm-torch/blob/mace_tutorial/tutorials/openmm-torch-MACE.ipynb

**Reply by svandenhaute** [reply]
OK, thanks! Our systems are periodic so I guess we'll have to wait it out. Really looking forward to the OpenMM interface though!

**Reply by ilyes319** [reply]
Nice to hear you will try out mace! You can open an issue so that you can be notified when periodic openMM is working.

**Reply by hityingph** [reply]
Hi, I'm eagerly awaiting the GPU version of the LAMMPS interface too. I've experimented with multiple MLP frameworks, like forward NN and GNN, for my current project. MACE has demonstrated the highest accuracy! I'm excited to conduct MD simulations with MACE for systems containing several thousand atoms. Thus, I'd like to check on the progress of the LAMMPS GPU interface one more time.
